tests/Behat/Context/Ui/GroupsContext.php

```diff
@@ -248,31 +166,35 @@ public function iShouldBeNotifiedThatItHasBeenSuccessfullyEdited(): void
     }
 
     /**
-     * @When I set its :attribute attribute to :value
-     * @When I set its :attribute attribute to :value in :language
-     * @When I do not set its :attribute attribute in :language
+     * @When I open :tab tab
      */
-    public function iSetItsAttributeTo($attribute, $value = null, $language = 'en_US'): void
+    public function iOpenTab(string $tabName): void
     {
-        $this->updateSimpleProductPage->addAttributeValue($attribute, $value ?? '', $language);
+        $this->updateSimpleProductPage->openTab($tabName);
     }
 
     /**
-     * @Then attribute :attributeName of product :product should be :value
-     * @Then attribute :attributeName of product :product should be :value in :language
+     * @When I want to modify the :productName product
      */
-    public function itsAttributeShouldBe($attributeName, ProductInterface $product, $value, $language = 'en_US'): void
+    public function iWantToModifyAProduct(string $productName): void
     {
-        Assert::same($this->updateSimpleProductPage->getAttributeValue($attributeName, $language), $value);
+        $product = $this->productRepository->findOneByCode($productName);
+
+        $this->sharedStorage->set('product', $product);
+        if ($product->isSimple()) {
+            $this->testHelper->waitUntilPageOpens($this->baseUpdateSimpleProductPage, ['id' => $product->getId()]);
+
+            return;
+        }
+
+        $this->testHelper->waitUntilPageOpens($this->baseUpdateSimpleProductPage, ['id' => $product->getId()]);
```

Comment:
What's the difference between the two page openings?

Reply:
changed